_posts/2021-05-05-deterministic-salts-in-namecoin-core.md

```diff
@@ -0,0 +1,24 @@
+---
+layout: post
+title: "Deterministic salts in Namecoin Core"
+author: yanmaani
+tags: [News]
+---
+
+Namecoin Core will, starting version 22.0, no longer require that a salt or TXID be provided in the `name_firstupdate` RPC call. If no transaction ID is provided, the wallet will perform a linear scan over its unspent outputs to attempt to find a matching transaction. If no salt is provided, it will assume that it can be deterministically generated from the private key using the same scheme already [implemented in Electrum-NMC]({{ "/2020/05/10/deterministic-salts.html" | relative_url }}).
+
+To facilitate this, `name_new` has also been changed to use deterministically generated salts.
+
+The old API is still supported; users are free to manually enter a salt, TXID, both, or neither. In fact, for `name_new` transactions created by versions of Namecoin Core prior to this change, the salt is still required. However, the transaction ID can always be determined automatically. In principle, it should only be necessary to provide the transaction ID to select which of several possible `name_new` outputs should be spent.
+
+There is therefore no longer any need to write down the salt and transaction ID going forward. With the new API, registering a name from the RPC console works like this:
+
+```
+name_new "d/myname"
+(wait 12 blocks...)
+name_firstupdate "d/myname" "value"
```

Comment:
Can we remove the `"value"` argument from this example?  That argument is optional now, and removing it would emphasize the simplicity of the API.

Reply:
Could it be marked as optional?

`name_firstupdate "d/myname" [value]`

Or if the audience is not familiar with the notation:
`name_firstupdate "d/myname" [value, optional]`

Reply:
I suspect that would be confusing.  If you really want to keep the `value` argument there, maybe just add a sentence below the code block mentioning that the `value` argument can also be omitted?

Reply:
Sure, done.